### ISSUE TITLE
We could make the automatic naming even a bit better

### ISSUE THREAD

**Issue by Evertt** [sahandnayebaziz/Hypertext#11]
So now the automatic naming of the library just literally takes the class name of the tag, which is cool for all the default html elements, but that doesn't work for custom elements like web-components. Custom elements are required to have a least one dash `-` in their name according to the html spec, AFAIK. So for example `<material-button></material-button>`. And how would someone name the class that represents that tag? I reckon that class would be called `MaterialButton` or `materialButton`.

With the current state of the library, if the developer would write this:

```swift
class materialButton: tag {}
```

Then their button would be called `<materialButton></materialButton>` instead of `<material-button></material-button>`. So how about we change the automatic naming to change camelCase class names to slug-case element names? I think it shouldn't be too hard.

**Comment by sahandnayebaziz** [comment]
I think this is a good idea. It's important to me that we keep the `name` property present, public, and easy to override. That way we can make this improvement without, for example, making it impossible to do camel case (if somebody wanted to, for whatever reason.) So since name can be overridden, I see no downside to this.

**Comment by sahandnayebaziz** [comment]
I'm going to come back to this after working on #9 and #10 :)

**Comment by Evertt** [comment]
Cool, I already found the way to do this:

```swift
extension String {
    func snaked(delimiter: String = "_") -> String {
        let range = NSMakeRange(0, self.characters.count)
        let regex = try! NSRegularExpression(pattern: "(.)(?=[A-Z])", options: [])
        return regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: "$1\(delimiter)").lowercased()
    }
}

print("MaterialButton".snaked(delimiter: "-")) // this now prints "material-button"
print("materialButton".snaked(delimiter: "-")) // also works if the first letter is lower cased
```

**Comment by sahandnayebaziz** [comment]
This was implemented in #23. Thank you @stupergenius :)
